### ISSUE TITLE
Document plot handlers.

### ISSUE THREAD

**Issue by pfebrer** [zerothi/sisl#838]
Methods like `Hamiltonian.plot.pdos()` are not documented anywhere, and the only way of getting information about them is if your editor can show you the arguments and documentation of the function.

I want to document them, but I don't know how exactly. I have thought of modifying the class template for autosummary (https://github.com/zerothi/sisl/blame/main/docs/_templates/autosummary/class.rst) to check for the plot attribute and then add some documentation, but I don't know exactly how or if this is the best approach.

**Comment by pfebrer** [comment]
I don't know if it is possible to make sphinx document things like `plot.pdos` as a normal method of the `Hamiltonian`. I think this could be a reasonable approach.

Another idea that I have is to gather all plot handlers and create a page in the `sisl.viz` documentation like:

```
List of objects that can be used to generate each plot:

AtomicMatrixPlot
================
  - SparseCSR
  - SparseOrbital
  - SparseAtom
  - Hamiltonian

GeometryPlot
================
  - Geometry
  - fdfSileSiesta
  - pdosSileSiesta
  - tbtncSileTBtrans
  - cubeSile
  - pdbSile
  - xsfSile
  - xyzSile
  - asciiSileBigDFT
  - inSileFHIaims
  - gotSileGULP
  - hamiltonianSile
  - mdSileOpenMX
  - omxSileOpenMX
  - txtSileORCA
  - refSileScaleUp
  - restartSileScaleUp
  - aniSileSiesta
  - tshsSileSiesta
  - onlysSileSiesta
  - hsxSileSiesta
  - ncSileSiesta
  - structSileSiesta
  - xvSileSiesta
  - stdoutSileSiesta
  - deltancSileTBtrans
  - tbtavncSileTBtrans
  - phtncSilePHtrans
  - phtavncSilePHtrans
  - tbtprojncSileTBtrans
  - phtprojncSilePHtrans
  - tbtsencSileTBtrans
  - phtsencSilePHtrans
  - carSileVASP
  - chgSileVASP
  - locpotSileVASP
  - winSileWannier90
  - tbSileWannier90
  - centresSileWannier90

GridPlot
================
  - Grid
  - fdfSileSiesta
  - cubeSile
  - ncSileSiesta
  - chgSileVASP
  - locpotSileVASP
  - rhoSileSiesta
  - ldosSileSiesta
  - rhoinitSileSiesta
  - rhoxcSileSiesta
  - drhoSileSiesta
  - baderSileSiesta
  - iorhoSileSiesta
  - totalrhoSileSiesta
  - stsSileSiesta
  - stmldosSileSiesta
  - hartreeSileSiesta
  - neutralatomhartreeSileSiesta
  - totalhartreeSileSiesta
  - gridncSileSiesta
  - tsvncSileSiesta

SitesPlot
================
  - BrillouinZone

BandsPlot
================
  - BrillouinZone
  - Path
  - fdfSileSiesta
  - wfsxSileSiesta
  - BandsData
  - Dataset
  - DataArray
  - bandsSileSiesta
  - Aiida_node

FatbandsPlot
================
  - BrillouinZone
  - wfsxSileSiesta
  - BandsData

PdosPlot
================
  - PDOSData
  - Path
  - fdfSileSiesta
  - pdosSileSiesta
  - tbtncSileTBtrans
  - Hamiltonian
  - wfsxSileSiesta
  - tbtavncSileTBtrans
  - phtncSilePHtrans
  - phtavncSilePHtrans
  - tbtprojncSileTBtrans
  - phtprojncSilePHtrans

WavefunctionPlot
================
  - Path
  - fdfSileSiesta
  - Hamiltonian
  - wfsxSileSiesta
  - EigenstateData
  - EigenstateElectron
```

with links to the corresponding methods.

This information could also be at the beggining of the showcase notebooks.

**Comment by pfebrer** [comment]
I have found a way to get method names with dots:

[![Screenshot from 2024-10-04 18-48-24](https://github.com/user-attachments/assets/225f2785-7f4a-4004-a8ee-ccd472b3c67b)](file:///home/pfebrer/Pictures/Screenshot%20from%202024-10-04%2018-48-24.png)

which I think is the most intuitive way for the user.

But to achieve this (after hours of trying) I have needed a dirty hack:
- Name the methods with a placeholder: E.g. "plot__SISLDOCSDOTPLACEHOLDER__pdos".
- On the `build_docs.sh` script, after building, replace the placeholder by a dot on all html files: `find . -type f -name '*.html' -exec sed -i 's/__SISLDOCSDOTPLACEHOLDER__/./g' {} +`

If you have a better idea of how to acheive this rendering...

**Comment by pfebrer** [comment]
Forgot to mention that it also has to show as a dot on the method documentation:

![Screenshot from 2024-10-04 18-57-21](https://github.com/user-attachments/assets/3e4a8439-c612-4a2a-843b-d30f47bca1fb)

**Comment by zerothi** [comment]
Isn't this resolved through #839?

**Comment by pfebrer** [comment]
Yes!
